bundles/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/handler/KonnectedHandler.java

```diff
@@ -288,7 +289,9 @@ public void initialize() {
         } catch (ConfigValidationException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
         }
-        this.moduleIpAddress = this.getThing().getProperties().get(HOST).toString();
+        if (baseUri == null) {
+            this.baseUri = this.getThing().getProperties().get(BASE_URL);
+        }
```

Comment:
New code to be removed, this is not a property but a configuration parameter.
You are getting the parameter and you set the member `baseUrl` in method `checkConfiguration`.
What you could/should do is to check in `checkConfiguration` that the config parameter is not null and if it is, you should trigger a `ConfigValidationException` exception

Reply:
As finally mentioned in another comment, you probably do not need to add any check.

Just remove these useless lines ad `baseUri` is set with a non null value in `checkConfiguration`.

Reply:
Thanks for the feedback. I did not realize the properties from the discovery would carry over to the configuration. I simplified the code a bit.